src/coreclr/vm/codeman.h

```diff
@@ -1318,13 +1373,13 @@ class ExecutionManager
     // infrastructure to manage readers so we can lock them out and delete domain data
     // make ReaderCount volatile because we have order dependency in READER_INCREMENT
 #ifndef DACCESS_COMPILE
-    static Volatile<RangeSection *> m_CodeRangeList;
-    static Volatile<LONG>   m_dwReaderCount;
-    static Volatile<LONG>   m_dwWriterLock;
+    static volatile RangeSectionHandleHeader * m_RangeSectionHandleReaderHeader;
+    static volatile RangeSectionHandleHeader * m_RangeSectionHandleWriterHeader;
```

Comment:
Plain volatile is not safe to use within our codebase. It has subtly different meanings on various compilers. Please either use the Volatile<T> type, or use explicit VolatileLoad/VolatileStore apis.